tests/templates/test_subroutines/test_qubitization.py

```diff
@@ -0,0 +1,335 @@
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize(
+        "use_jit , shots",
+        ((False, None), (True, None), (False, 50000)),
+    )  # TODO: (True, 50000) fails because jax.jit on jax.grad does not work with AmplitudeEmbedding
+    def test_qnode_jax(self, shots, use_jit):
+        """ "Test that the QNode executes and is differentiable with JAX. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        import jax
+
+        jax.config.update("jax_enable_x64", True)
+
+        dev = qml.device("default.qubit", shots=shots, seed=10)
+        diff_method = "backprop" if shots is None else "parameter-shift"
+        qnode = qml.QNode(self.circuit, dev, interface="jax", diff_method=diff_method)
+        if use_jit:
+            qnode = jax.jit(qnode)
+
+        params = jax.numpy.array(self.params)
+
+        jac_fn = jax.jacobian(qnode)
+        if use_jit:
+            jac_fn = jax.jit(jac_fn)
+
+        jac = jac_fn(params)
+        assert jac.shape == (4,)
+        assert np.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize(
+        "shots", [None]
+    )  # TODO: finite shots fails because Prod is not currently differentiable.
+    def test_qnode_torch(self, shots):
+        """ "Test that the QNode executes and is differentiable with Torch. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        import torch
+
+        dev = qml.device("default.qubit", shots=shots, seed=10)
+        diff_method = "backprop" if shots is None else "parameter-shift"
+        qnode = qml.QNode(self.circuit, dev, interface="torch", diff_method=diff_method)
+
+        params = torch.tensor(self.params, requires_grad=True)
+        jac = torch.autograd.functional.jacobian(qnode, params)
+        assert qml.math.shape(jac) == (4,)
+        assert qml.math.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("shots", [None, 50000])
+    @pytest.mark.xfail(reason="tf gradient doesn't seem to be working, returns ()")
+    def test_qnode_tf(self, shots):
+        """ "Test that the QNode executes and is differentiable with TensorFlow. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit", shots=shots, seed=10)
+        diff_method = "backprop" if shots is None else "parameter-shift"
+        qnode = qml.QNode(self.circuit, dev, interface="tf", diff_method=diff_method)
+
+        params = tf.Variable(self.params)
+        with tf.GradientTape() as tape:
+            res = qnode(params)
+
+        jac = tape.gradient(res, params)
+        assert qml.math.shape(jac) == (4,)
+        assert qml.math.allclose(res, self.exp_grad, atol=0.001)
+
+    @pytest.mark.xfail(reason="see https://github.com/PennyLaneAI/pennylane/issues/5507")
```

Comment:
I am happy with merging this as is 👍🏼, we can make a shortcut ticket to fix this once the Sum has better grad support. We could also make a separate PR to do the work-around 👍🏼

Reply:
Added a note in the issue :)